File: ips/ips-model-analysis/src/main/java/io/iochord/apps/ips/model/analysis/services/ism/IsmDiscoveryConfiguration.java
package io.iochord.apps.ips.model.analysis.services.ism;

import java.util.Map;

import lombok.Getter;
import lombok.Setter;

/**
*
* @package ips-model-analysis
* @author Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
* @since 2019
*
*/
public class IsmDiscoveryConfiguration {

	@Getter
	@Setter
	private String datasetId;

	@Getter
	@Setter
	private String colCaseId = "ci";

	@Getter
	@Setter
	private String colEventActivity = "ea";

	@Getter
	@Setter
	private String colEventTimestamp = "es";

	@Getter
	@Setter
	private String colEventTimeStart = "es";

	@Getter
	@Setter
	private String colEventTimeComplete = "ec";

	@Getter
	@Setter
	private int skipRows = 1;

	@Getter
	@Setter
	private int positiveObservationThreshold = 0;
	
	@Getter
	@Setter
	private float dependencyThreshold = -1f;
	
	@Getter
	@Setter
	private Map<String, String> connectorPaths;
	
	@Getter
	@Setter
	private double animatorLength = 60;
	
	@Getter
	@Setter
	private int tokenMinSize = 5;
	
	@Getter
	@Setter
	private int tokenMaxSize = 10;

}

File: ips/ips-simulator-web/src/main/ts/src/iochord/ips/analysis/process-discovery/models/IsmDiscoveryConfiguration.ts
/**
 * Case mapping settings for ISM discovery API.
 *
 * @export
 * @class IsmDiscoveryConfiguration
 *
 * @package ts
 * @author Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
 * @since 2019
 */
export default class IsmDiscoveryConfiguration {

  /**
   * The dataset id of the event log.
   *
   * @type {string}
   * @memberof IsmDiscoveryConfiguration
   */
  public datasetId: string = '';

  /**
   * The column position for case id.
   *
   * @type {string}
   * @memberof IsmDiscoveryConfiguration
   */
  public colCaseId: string = 'ci';

  /**
   * The column position for event activity.
   *
   * @type {string}
   * @memberof IsmDiscoveryConfiguration
   */
  public colEventActivity: string = 'ea';

  /**
   * The column position for event timestamp.
   *
   * @type {string}
   * @memberof IsmDiscoveryConfiguration
   */
  public colEventTimestamp: string = 'ec';

  public positiveObservationThreshold: number = 0;

  /**
   * The column position for default dependency threshold
   *
   * @type {number}
   * @memberof IsmDiscoveryConfiguration
   */
  public dependencyThreshold: number = -1;

  public connectorPaths: any = {};

  public animatorLength: number = 30;

  public tokenReplay: boolean = false;

  public tokenReplayDecorateLabel: boolean = false;
}
